internal/commands/worker/config.go

```diff
@@ -168,6 +139,42 @@ func buildWorker(cmd *cobra.Command) (*worker.Worker, error) {
 
 	opts = append(opts, worker.WithLogger(logger))
 
+	// Configure upstreams
+	if len(config.Upstreams) == 0 {
+		config.Upstreams = append(config.Upstreams, ConfigUpstream{
+			Token:    config.Token,
+			Endpoint: config.RPC.Endpoint,
+		})
+	} else {
+		if config.Token != "" {
+			return nil, fmt.Errorf("%w: \"token:\" and \"endpoints:\" are mutually exclusive",
+				ErrConfiguration)
+		}
+
+		if config.RPC.Endpoint != upstream.DefaultRPCEndpoint {
+			return nil, fmt.Errorf("%w: \"rpc:\" and \"endpoints:\" are mutually exclusive",
+				ErrConfiguration)
+		}
+	}
+
+	for _, configUpstream := range config.Upstreams {
```

Comment:
Probably can also add a validation that only resource aware workers can have more than 1 upstream.